api/models/build.js

```diff
@@ -79,10 +76,10 @@ const completeJobSiteUpdate = (build, completedAt) => {
   return Promise.resolve();
 };
 
-function completeJob(err) {
+function updateJobStatus(buildStatus) {
   const completedAt = new Date();
 
-  return completeJobStateUpdate(err, this, completedAt)
+  return jobStateUpdate(buildStatus, this, completedAt)
     .then(build => completeJobSiteUpdate(build, completedAt)
       .then(() => build));
 }
```

Comment:
Now or as a future PR, consider refactoring a bit to be a little more clear, would love something like:
```
async function updateJobStatus(buildStatus) {
  const completedAt = new Date();
  const build = await jobStateUpdate(buildStatus, this, completedAt);
  await completeJobSiteUpdate(build, completedAt);
  return build;
}
```

Also consider moving the conditional logic if the status is "success" into this method and out of `completeJobSiteUpdate` Ex.
```
async function updateJobStatus(buildStatus) {
  const completedAt = new Date();
  const build = await jobStateUpdate(buildStatus, this, completedAt);
  if (buildStatus.state === 'success') {
    await completeJobSiteUpdate(build, completedAt);
  }
  return build;
}
```

then the job of `completeJobSiteUpdate` becomes much simpler and does one thing and always returns the site.